test_unstructured/partition/test_pdf.py

```diff
@@ -132,3 +132,15 @@ def test_partition_pdf(url, api_called, local_called):
         pdf.partition_pdf(filename="fake.pdf", url=url)
         assert pdf._partition_via_api.called == api_called
         assert pdf._partition_pdf_or_image_local.called == local_called
+
+
+@pytest.mark.parametrize(
+    "url, api_called, local_called", [("fakeurl", True, False), (None, False, True)]
+)
+def test_partition_pdf_with_template(url, api_called, local_called):
+    with mock.patch.object(
```

Comment:
This looks good, but fore future reference`pytest.mocker` and/or `pytest.monkeypatch` is generally preferred over `with mock.patch...`